tests/urlpatterns_reverse/tests.py

```diff
@@ -1758,3 +1758,11 @@ def test_invalid_reverse(self):
             with self.subTest(name=name, kwargs=kwargs):
                 with self.assertRaises(NoReverseMatch):
                     reverse(name, kwargs=kwargs)
+
+@override_settings(ROOT_URLCONF="urlpatterns_reverse.urls")
+class TestRereverse(TestCase):
+    def test_rereverse(self):
+        match = resolve("/rereverse/12/")
+        self.assertEqual(reverse(match.url_name, args=match.args, kwargs=match.kwargs), '/rereverse/12/')
+        match = resolve('/rereverse-overridden/12/url/')
+        self.assertEqual(reverse(match.url_name, args=match.args, kwargs=match.kwargs), '/rereverse-overridden/12/url/')
```

Comment:
Hi @apollo13, So I made the changes you requested and was able to implement all test cases except one, test case where 
```
match = resolve('/rereverse-overridden/12/url/')
self.assertEqual(reverse(match.url_name, args=match.args, kwargs=match.kwargs), '/rereverse-overridden/12/url/')
```
Now, I've tried my best to fix it but this test keeps on failing cause of  
```
AssertionError: '/rereverse-overridden/12/default/' != '/rereverse-overridden/12/url/'
```
even when I tried to force-feed the different kwargs, it failed.
```
Traceback (most recent call last):
  File "C:\django\git\django\tests\urlpatterns_reverse\tests.py", line 1769, in test_rereverse
    self.assertEqual(reverse(match.url_name, args=match.args, kwargs=match.kwargs), '/rereverse-overridden/12/url/')
  File "c:\django\git\django\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
  File "c:\django\git\django\django\urls\resolvers.py", line 828, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'rereverse-overridden' with keyword arguments '{'arg': 12, 'overridden': 'url'}' not found. 1 pattern(s) tried: ['rereverse-overridden/(?P<arg>\\d+)/(?P<overridden>\\w+)/$']

----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\django\git\django\tests\urlpatterns_reverse\tests.py", line 1769, in test_rereverse
    self.assertEqual(reverse(match.url_name, args=match.args, kwargs=match.kwargs), '/rereverse-overridden/12/url/')
  File "c:\django\git\django\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
  File "c:\django\git\django\django\urls\resolvers.py", line 828, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'rereverse-overridden' with keyword arguments '{'arg': 12, 'overridden': 'url', 'extra': True}' not found. 1 pattern(s) tried: ['rereverse-overridden/(?P<arg>\\d+)/(?P<overridden>\\w+)/$']
```

Now, if you could please at least can you help me understand what I am overlooking here.
I would be grateful if you could just give me some hint of what I am missing.

thanks!